Gems/AtomLyIntegration/AtomRenderOptions/Code/Source/AtomRenderOptionsActionHandler.cpp

```diff
@@ -0,0 +1,122 @@
+ *
+ */
+
+#include "AtomRenderOptionsActionHandler.h"
+
+#include "AtomRenderOptions.h"
+
+#include <AzCore/Debug/Trace.h>
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/Name/Name.h>
+
+#include <AzToolsFramework/ActionManager/Action/ActionManagerInterface.h>
+#include <AzToolsFramework/ActionManager/Menu/MenuManagerInterface.h>
+#include <AzToolsFramework/ActionManager/ToolBar/ToolBarManagerInterface.h>
+#include <AzToolsFramework/Editor/ActionManagerIdentifiers/EditorContextIdentifiers.h>
+#include <AzToolsFramework/Editor/ActionManagerIdentifiers/EditorToolBarIdentifiers.h>
+
+namespace AZ::Render
+{
+    constexpr AZStd::string_view ViewportRenderOptionsMenuIdentifier = "o3de.menu.editor.viewport.renderoptions";
+
+    void AtomRenderOptionsActionHandler::Activate()
+    {
+        m_actionManagerInterface = AZ::Interface<AzToolsFramework::ActionManagerInterface>::Get();
+        m_menuManagerInterface = AZ::Interface<AzToolsFramework::MenuManagerInterface>::Get();
+        m_toolBarManagerInterface = AZ::Interface<AzToolsFramework::ToolBarManagerInterface>::Get();
+
+        if (m_actionManagerInterface && m_menuManagerInterface && m_toolBarManagerInterface)
+        {
+            AzToolsFramework::ActionManagerRegistrationNotificationBus::Handler::BusConnect();
+        }
+
+        AzToolsFramework::EditorEventsBus::Handler::BusConnect();
+    }
+
+    void AtomRenderOptionsActionHandler::Deactivate()
+    {
+        AzToolsFramework::EditorEventsBus::Handler::BusDisconnect();
+        AzToolsFramework::ActionManagerRegistrationNotificationBus::Handler::BusDisconnect();
+    }
+
+    void AtomRenderOptionsActionHandler::OnMenuRegistrationHook()
+    {
+        AzToolsFramework::MenuProperties menuProperties;
+        menuProperties.m_name = "Render Options";
+        m_menuManagerInterface->RegisterMenu(ViewportRenderOptionsMenuIdentifier, menuProperties);
+    }
+
+    void AtomRenderOptionsActionHandler::OnActionRegistrationHook()
+    {
+        // Render Options menu icon
+        {
+            constexpr AZStd::string_view actionIdentifier = "o3de.action.viewport.renderoptions";
+            AzToolsFramework::ActionProperties actionProperties;
+            actionProperties.m_name = "Render Options";
+            actionProperties.m_iconPath = ":/Icons/Material_80.svg";
+
+            m_actionManagerInterface->RegisterAction(
+                EditorIdentifiers::MainWindowActionContextIdentifier,
+                actionIdentifier,
+                actionProperties,
+                []
+                {
+                });
+        }
+
+        // Temporal anti aliasing
+        {
+            constexpr AZStd::string_view actionIdentifier = "o3de.action.viewport.renderoptions.taa";
+            AzToolsFramework::ActionProperties actionProperties;
+            actionProperties.m_name = "Anti-aliasing (TAA)";
+            actionProperties.m_description = "Use temporal anti-aliasing";
```

Comment:
TAA won't necessarily be available in every pipeline or project. Ideally, this should loop over all of the available passes and add them to the menu. Hopefully, there is an easy, generic way to enumerate the passes, loop over them, get their names, create the action identifiers, and register them with the menu.

Reply:
That's a great idea :) I'll look into it